ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/SchemaHelper.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.auth;
+
+import java.net.InetAddress;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.gms.ApplicationState;
+import org.apache.cassandra.gms.EndpointState;
+import org.apache.cassandra.gms.Gossiper;
+import org.apache.cassandra.gms.VersionedValue;
+import org.apache.cassandra.utils.FBUtilities;
+
+class SchemaHelper
+{
+    private static final Logger LOG = LoggerFactory.getLogger(SchemaHelper.class);
+
+    private final InetAddress localAddress;
+    private final Gossiper gossiper;
+
+    SchemaHelper()
+    {
+        this(FBUtilities.getBroadcastAddress(), Gossiper.instance);
+    }
+
+    @VisibleForTesting
+    SchemaHelper(InetAddress localAddress, Gossiper gossiper)
+    {
+        this.localAddress = localAddress;
+        this.gossiper = gossiper;
+    }
+
+    void waitForSchemaAlignment(int maxTimeToWaitInSeconds)
```

Comment:
Would it make sense if this method returned a boolean so that the caller could log a warning if schema agreement was not reached? (Could be used to more easily determine cause of exceptions)